### PR TITLE
[refactor](functions) Check ReturnType of functions on FE

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

will update for checking other functions weather need to apply such a check on FE.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/FunctionCallExpr.java [hunk 1]
```diff
@@ -2047,15 +2053,29 @@ protected boolean isConstantImpl() {
...
         if (fnName.equalsIgnoreCase("sleep")) {
             return false;
         }
-        return super.isConstantImpl();
+
+        return super.isConstantImpl(); // True as default.
+    }
+
+    private boolean isConstantForSpecialCases(String fnName) {
+        if (fnName.equalsIgnoreCase("is_null_pred") || fnName.equalsIgnoreCase("is_not_null_pred")) {
```
Comment (BiteTheDDDDt) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/FunctionCallExpr.java:2073]: is_null_pred's is_constant depending on argument column's is_nullable.

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/FunctionCallExpr.java [hunk 2]
```diff
@@ -2047,15 +2048,29 @@ protected boolean isConstantImpl() {
...
             return false;
         }
-        // Sleep is a special function for testing.
+        // specially decide some functions.
         if (fnName.equalsIgnoreCase("sleep")) {
             return false;
         }
-        return super.isConstantImpl();
+
+        return super.isConstantImpl(); // True as default.
```
Comment (zhangstar333) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/FunctionCallExpr.java:2064]: here seems has no relationship between result nullable and children nullable,

**Pull request closed without merging**